File: rb/features.rb
# FinalSpace SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FinalSpaceFeatures
  def self.make_feature(name)
    case name
    when "base"
      FinalSpaceBaseFeature.new
    when "ratelimit"
      FinalSpaceRatelimitFeature.new
    when "retry"
      FinalSpaceRetryFeature.new
    when "test"
      FinalSpaceTestFeature.new
    when "timeout"
      FinalSpaceTimeoutFeature.new
    else
      FinalSpaceBaseFeature.new
    end
  end
end

File: py/features.py
# FinalSpace SDK feature factory

from feature.base_feature import FinalSpaceBaseFeature
from feature.test_feature import FinalSpaceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FinalSpaceBaseFeature(),
        "test": lambda: FinalSpaceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
